cosmovisor/process.go

```diff
@@ -142,3 +146,28 @@ func doBackup(cfg *Config) error {
 
 	return nil
 }
+
+// Runs the pre-upgrade command defined by the application
+func doPreUpgrade(cfg *Config) error {
+	bin, err := cfg.CurrentBin()
+	preUpgradeCmd := exec.Command(bin, "pre-upgrade", "--unsafe-skip-upgrades")
```

Comment:
Also we need to read `--unsafe-skip-upgrades` from env and append here (if not nil)

Reply:
Oh not exactly. `--unsafe-skip-upgrades` is not needed for `$DAEMON pre-upgrade`, it is a flag for `$DAEMON start` command. Here we need to check if skip-upgrades is set, so we should not run `pre-upgrade` if it is set

Reply:
@anilCSE  You mean when `--unsafe-skip-upgrade` is set `pre-upgrade` should not be run from the cosmovisor?

Reply:
Yes